src/EitherAsync.ts

```diff
@@ -111,6 +113,12 @@ class EitherAsyncImpl<L, R> implements EitherAsync<L, R> {
     })
   }
 
+  swap(): EitherAsync<R, L> {
+    return EitherAsync(async (helpers) => {
+      throw await this.runPromise((helpers as any) as EitherAsyncHelpers<L>)
```

Comment:
I am quite surprise by this implementation (inspired from mapLeft), but the tests passed with this...

Reply:
This doesn't seem right, I'll have to check it out on my machine.

Reply:
Turned out the tests for Either are not working at all. `Left` and `Right` have the same JSON value, so Jest can't disambiguate between them.